pos_geonames/static/src/js/pos_geonames_field.js

```diff
@@ -0,0 +1,107 @@
+        }
+
+        captureChange(event) {
+            let country_id_before = parseInt(this.changes.country_id);
+            this.changes[event.target.name] = event.target.value;
+            if (country_id_before !== parseInt(this.changes.country_id)) {
+                $('.client-address-states').empty();
+                $('.client-address-states').append(new Option("None", ""))
+                this.env.pos.states.forEach((state) => {
+                    if (state.country_id[0] === parseInt(this.changes.country_id)) {
+                        $('.client-address-states').append(new Option(state.name, state.id))
+                    }
+                })
+            }
+        }
+
+        _updateStatesList(data) {
+            let country_id_before = parseInt(this.changes.country_id);
+            this.changes.country_id = data;
+            this.el.getElementsByClassName("client-address-country")[0].value = data;
+            if (country_id_before !== parseInt(this.changes.country_id)) {
+                $('.client-address-states').empty();
+                $('.client-address-states').append(new Option("None", ""))
+                this.env.pos.states.forEach((state) => {
+                    if (state.country_id[0] === parseInt(this.changes.country_id)) {
+                        $('.client-address-states').append(new Option(state.name, state.id))
+                    }
+                })
+            }
+        }
+
+        /**
+         * @private
+         * @param {KeyBoardEvent} ev
+         */
+        async _updateCountryList(ev) {
+            const self = this;
+            const query = self.choices.input.element.value
+            rpc.query({
+                model: 'res.city.zip',
+                method: 'search_read',
+                kwargs: {
+                    fields: ['id', 'display_name', 'name', 'city_id', 'country_id', 'state_id'],
+                    domain: [
+                        ['display_name', 'ilike', query]
+                    ],
+                    limit: 25
+                }
+            }).then(function (data) {
+                    let countries = [];
+                    _.each(data, function (country) {
+                        countries.push({
+                            value: country.id,
+                            label: country.display_name,
+                            customProperties: {
+                                zip: country.name,
+                                state: country.state_id[0],
+                                country: country.country_id[0],
+                                city: country.city_id[1],
+                            }
+                        })
+                    });
+                    self.choices.setChoices(
+                        countries,
+                        'value',
+                        'label',
+                        true
+                    );
+                }
+            );
```

Comment:
```suggestion
            const query = self.choices.input.element.value;
            const data = await rpc.query({
                model: 'res.city.zip',
                method: 'search_read',
                kwargs: {
                    fields: ['id', 'display_name', 'name', 'city_id', 'country_id', 'state_id'],
                    domain: [
                        ['display_name', 'ilike', query]
                    ],
                    limit: 25
                }
            })
            
            countries = data.map((rec) => {
                return {
                  value: rec.id,
                  label: rec.display_name.
                  customProperties: {
                    zip: rec.name,
                    state: rec.state_id[0],
                    country: rec.country_id[0],
                    city: rec.city_id[1]
                }
            });

            this.choices.setChoices(countries, 'value', 'label', true);
```

Suggestion: it's the same but written in modern js, making use of `async`/`await`, and arrow functions to get rid of `const self = this;` and the callback hell